pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -95,6 +96,28 @@ func handleVolumeDelete(ctxArg interface{}, key string,
 	log.Functionf("handleVolumeDelete(%s) Done", key)
 }
 
+// useZVolDisk returns true if we should use zvol for the provided VolumeStatus
+func useZVolDisk(ctx *volumemgrContext, status *types.VolumeStatus) bool {
+	if status.IsContainer() {
+		return false
+	}
```

Comment:
The old code doesn't have this check, but appears to create a zvol in this case. 
But do we never call createVdiskVolume() when IsContainer? If not then it would make more sense to have a check at the top of createVdiskVolume() and log.Fatal if IsContainer.